map/set-max-extent/src/main/java/com.esri.samples.set_max_extent/SetMaxExtentSample.java

```diff
@@ -0,0 +1,142 @@
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class SetMaxExtentSample extends Application {
+
+    private MapView mapView;
+
+
+    @Override
+    public void start(Stage stage) {
+
+        try {
+            // create stack pane and application scene
+            var stackPane = new StackPane();
+            var scene = new Scene(stackPane);
+
+            // set title, size, and add scene to stage
+            stage.setTitle("Set Max Extent Sample");
+            stage.setWidth(800);
+            stage.setHeight(700);
+            stage.setScene(scene);
+            stage.show();
+
+            // authentication with an API key or named user is required to access basemaps and other location services
+            String yourAPIKey = System.getProperty("apiKey");
+            ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+            // create a map with the streets basemap style
+            var map = new ArcGISMap(BasemapStyle.ARCGIS_STREETS);
+
+            // create a map view and set the map to it
+            mapView = new MapView();
+            mapView.setMap(map);
+
+            // create a max extent envelope with points on Colorado's northwest and southeast corners
+            var coloradoNW = new Point(-12139393.2109, 5012444.0468, SpatialReferences.getWebMercator());
+            var coloradoSE = new Point(-11359277.5124, 4438148.7816, SpatialReferences.getWebMercator());
+            var extentEnvelope = new Envelope(coloradoNW, coloradoSE);
+
+            // set the map's max extent to the max extent envelope
+            map.setMaxExtent(extentEnvelope);
+
+            // create a graphics overlay of the map's max extent and add it to the map view
+            var coloradoGraphicsOverlay = new GraphicsOverlay();
+            coloradoGraphicsOverlay.getGraphics().add(new Graphic(map.getMaxExtent()));
+            mapView.getGraphicsOverlays().add(coloradoGraphicsOverlay);
+
+            // create a simple renderer with a red dashed line and set it to be the renderer for the graphics overlay
+            var simpleRenderer = new SimpleRenderer(new SimpleLineSymbol(SimpleLineSymbol.Style.DASH, ColorUtil.colorToArgb(Color.RED), 5));
+            coloradoGraphicsOverlay.setRenderer(simpleRenderer);
+
+            // create a control panel for the on/off toggles
+            var controlsVBox = new VBox(6);
+            controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.4)"), CornerRadii.EMPTY,
+                    Insets.EMPTY)));
+            controlsVBox.setPadding(new Insets(10.0));
+            controlsVBox.setMaxSize(120, 75);
+
+            // create a toggle group for the enabled/disable buttons
+            var toggleGroup = new ToggleGroup();
+
+            // create the enable/disable buttons and add them to the toggle group
+            var toggleOn = new ToggleButton("Max Extent ON ");
+            var toggleOff = new ToggleButton("Max Extent OFF");
```

Comment:
There are a couple of gotchas with this toggle button approach that we would have to address. Given the code complexity we already have here, I'm loathe to suggest we add more code. On reflection I think going for a check box is going to be the better route here. Nonetheless this has been good experience for you to get to grips with a range of JavaFX UI components!